src/MemcacheCachePool.php

```diff
@@ -12,30 +12,41 @@
 namespace Cache\Adapter\Memcache;
 
 use Cache\Adapter\Common\AbstractCachePool;
+use Memcache;
 use Psr\Cache\CacheItemInterface;
 
-/**
- *
- */
 class MemcacheCachePool extends AbstractCachePool
 {
+    /**
+     * @var Memcache
+     */
+    private $cache;
+
+    public function __construct(Memcache $cache)
+    {
+        $this->cache = $cache;
+    }
+
     protected function fetchObjectFromCache($key)
     {
-        return false;
+        return $this->cache->get($key);
     }
 
     protected function clearAllObjectsFromCache()
     {
-        return true;
+        return $this->cache->flush();
     }
 
     protected function clearOneObjectFromCache($key)
     {
-        return true;
+        $flags = false;
+        $this->cache->delete($key, $flags);
+
+        return $flags === false;
```

Comment:
We do only have to return false on error. I think some other adapters always return true.